"Source" in problem panel can be misreported as " "_generated_diagnostic_collection_name_#0"

I've just had some errors reported using this very handy extension, but in the "Problems" panel in VS Code (when set to "View As Table") where it has the "Source" listed as `_generated_diagnostic_collection_name_#0` . Copying the message gives me:

```json
[{
	"resource": "/workspace/script.sh",
	"owner": "_generated_diagnostic_collection_name_#0",
	"code": "SC2178",
	"severity": 4,
	"message": "Variable was used as an array but is now assigned a string.",
	"source": "shellcheck",
	"startLineNumber": 414,
	"startColumn": 11,
	"endLineNumber": 414,
	"endColumn": 25
}]
```
(the "code" value is shown as `shellcheck(SC2154)`)

but I have seen it on others such as:
```json
[{
	"resource": "/workspace/script.sh",
	"owner": "_generated_diagnostic_collection_name_#0",
	"code": "SC2091",
	"severity": 4,
	"message": "Remove surrounding $() to avoid executing output (or use eval if intentional).",
	"source": "shellcheck",
	"startLineNumber": 36,
	"startColumn": 16,
	"endLineNumber": 36,
	"endColumn": 51
}]
```
and
```json
[{
	"resource": "/workspace/otherscript.sh",
	"owner": "_generated_diagnostic_collection_name_#0",
	"code": "SC2154",
	"severity": 4,
	"message": "test is referenced but not assigned (for output from commands, use \"$(test ...)\" ).",
	"source": "shellcheck",
	"startLineNumber": 20,
	"startColumn": 1,
	"endLineNumber": 20,
	"endColumn": 6
}]
```

I **believe** that (according [to the docs](https://code.visualstudio.com/docs/editor/tasks-appendix#:~:text=build%20output.%0A%20*%2F%0Ainterface-,problemmatcher%20,-%7B%0A%20%20%2F**%0A%20%20%20*%20The%20name%20of):
* the `owner` field should be `shellcheck`
* the `resource` field should not be set
* the filename should be in `file`

I think a regexp has just got updated and the pattern numbers haven't been updated to reflect the changes (but I've never written a VSCode Extension so take most of this with a grain of salt!).

**Platform details**
- Extension version: v0.19.6
- With shellcheck installed on a `Debian:bullseye` docker image (v 0.8.0) running as a DevContainer ( ` "shellcheck.executablePath": "/usr/local/bin/shellcheck"`)
- VS Code Version: 1.69.2 (user setup)
Electron: 18.3.5
Chromium: 100.0.4896.160
Node.js: 16.13.2
V8: 10.0.139.17-electron.0
OS: Windows_NT x64 10.0.19044

--- a/src/linter.ts
+++ b/src/linter.ts
@@ -101,7 +101,8 @@ export default class ShellCheckProvider implements vscode.CodeActionProvider {
     this.delayers = Object.create(null);
     this.settingsByUri = new Map();
     this.toolStatusByPath = new Map();
-    this.diagnosticCollection = vscode.languages.createDiagnosticCollection();
+    this.diagnosticCollection =
+      vscode.languages.createDiagnosticCollection("shellcheck");
     this.codeActionCollection = new Map();
     this.additionalDocumentFilters = new Set();
 

